Sources/MessagingPush/MessagingPush+AppDelegate.swift

```diff
@@ -84,7 +84,7 @@ extension MessagingPushImplementation {
         _ center: UNUserNotificationCenter,
         didReceive response: UNNotificationResponse
     ) -> CustomerIOParsedPushPayload? {
-        if sdkConfig.autoTrackPushEvents {
+        if moduleConfig.autoTrackPushEvents {
```

Comment:
🤔 I think this could be a problem for us. That is assuming that we want SDK wrapper customers to be able to use Dart and JS to configure push features: 

```js
// RN example
// in App.js file: 
let cioSdkConfig = ...
cioSdkConfig.autoTrackPushEvents = false 
CustomerIO.initialize(cioSdkConfig)
```

See "Risks" section in [this ticket](https://github.com/customerio/issues/issues/11728) where I try to explain this scenario and why this `if` statement living in the messaging push module would not work. Because this function would execute before RN/Flutter SDK initialize would execute. 

This might mean that the messagingpush module can still accept the `autoTrackPushEvents` config property, but it might need to somehow pass that option to another module like journeys or CDP to handle this logic? Or the `CustomerIO` class holds this config option and passes it to the journeys module? Not sure. 

This code here is code that we have in prod today. However, I know that one of our goals with this project is that config options live inside of feature modules. If we want `autoTrackPushEvents` to live inside of this module, that's why I am bringing this problem up.

Reply:
Thanks for sharing Levi. I do agree with the problem. However, at this point, a quicker solution that I can think of for customers using wrapper SDKs might be to add something like the following in their `AppDelegate` / `NotificationServiceExtension`:

```swift
MessagingPush.shared.config { config in 
    config.autoTrackPushEvents = false 
}
```

This obviously isn't final yet, and we'll continue to refine it as we work on the wrappers. But I think it could serve as a good short-term fix, while the [ticket](https://github.com/customerio/issues/issues/11728) you shared addresses the problem in the long run.

Regarding passing to Journeys, I don’t believe this falls within the responsibility of the Journey module. The Push module should evaluate this and only pass the tracking event to Journeys if the flag is enabled. The `autoTrackPushEvents` flag is specific to the Push module. Journeys/CDP shouldn’t need to interact with this flag; if they receive a metric event, they should simply track it.